performance_regression_gradual_grow_throughput.py

```diff
@@ -18,24 +19,58 @@ class CSPopulateDistribution(Enum):
     UNIFORM = "uniform"
 
 
+class Workload(NamedTuple):
+    workload_type: str
+    cs_cmd_tmpl: list
+    cs_cmd_warm_up: list | None
+    num_threads: int
+    throttle_steps: list
+    preload_data: bool
+    drop_keyspace: bool
+    wait_no_compactions: bool
+
+
 class PerformanceRegressionGradualGrowThroughputTest(PerformanceRegressionTest):  # pylint: disable=too-many-instance-attributes
+    """
+    This class presents new performance test that run gradual increased throughput steps.
+    The test run steps with different throughput.
+    Throughput of every step is fixed and defined hardcoded according to the load type (write, read
+    and mixed). Last step is unthrottled.
+    Latency for every step is received from cassandra-stress HDR file and reported in Argus and email.
+    """
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        # all parameters were taken from scylla-stress-orch repo
-        # Planned data size is 3 TB in total: 1tb per node
         self.CLUSTER_SIZE = self.params.get("n_db_nodes")  # pylint: disable=invalid-name
         self.REPLICATION_FACTOR = 3  # pylint: disable=invalid-name
 
+    def throttle_steps(self, workload_type):
+        throttle_steps = self.params["perf_gradual_throttle_steps"]
+        if workload_type not in throttle_steps:
+            TestFrameworkEvent(source=self.__class__.__name__,
+                               message=f"Throttle steps for '{workload_type}' test is not defined in "
+                                       f"'perf_gradual_throttle_steps' parameter",
+                               severity=Severity.CRITICAL).publish()
+        return throttle_steps
```

Comment:
isn't this should be ?
```suggestion
        return throttle_steps[workload_type] 
```

Reply:
right, fixed